db/version_set.cc

```diff
@@ -5768,7 +5770,7 @@ Status VersionSet::ProcessManifestWrites(
     for (auto v : versions) {
       delete v;
     }
-    if (manifest_io_status.ok()) {
+    if (manifest_io_status.ok() && !current_file_io_status.ok()) {
```

Comment:
What if s is set to non-ok after successfully creating the CURRENT file? Would it make more sense to just record if a new MANIFEST is created and check that flag here?

Reply:
In this case,  this keeps the `manifest_number_` unchanged, to point to the existing manifest and not updating its size.   If we crashed right after this and CURRENT points to the new manifest file, the new manifest file should be around to be used since below logic only remove the new manifest file if manifest write IO itself encountered issues.  If we didn't crash, this divergence will be fixed by the next manifest write attempt when it sees `descriptor_log_` is nullptr.

Anyways, in these error scenario, `manifest_number_` no longer points to what `descriptor_log_` points to,  I wonder should we err towards keeping it point to the old manifest number, or the new manifest number (knowing it is in a good state, because manifest_io_status is ok, but current file creation may have some issues). Do you have any concerns?

Reply:
I think either is fine. I found checking current_file_io_status harder to reason about since we are updating the manifest_file_number when CURRENT creation failed, but not updating it when CURRENT Is created successfully.

> wonder should we err towards keeping it point to the old manifest number (which is removing this code block altogether)

Yeah, something probably went wrong with the new MANIFEST. Maybe backup should read the old MANIFEST that's valid.

Reply:
Good point that in this approach, manifest number is updated when current file could encounter issues, looks very unintuitive. I have removed this block altogether.